conformance/utils/suite/experimental_suite.go

```diff
@@ -0,0 +1,278 @@
+//go:build experimental
+// +build experimental
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/conformance"
+	confv1a1 "sigs.k8s.io/gateway-api/conformance/apis/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/utils/config"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Test Suite - Public Types
+// -----------------------------------------------------------------------------
+
+// ConformanceTestSuite defines the test suite used to run Gateway API
+// conformance tests.
+type ExperimentalConformanceTestSuite struct {
```

Comment:
As part of this PR, I think it would be helpful to have some documentation for how to actually run these tests. For example, it's not obvious to me how I'd fill in the implementation information below.

Reply:
I added some indications in the code as comments. Were you thinking about something more structured, or is it enough? 76dcc19